terraform/modules/tf_stream_alert/iam.tf

```diff
@@ -18,9 +18,17 @@ data "aws_iam_policy_document" "lambda_assume_role_policy" {
   }
 }
 
+// IAM Role Policy: Allow the Rule Processor to write CloudWatch logs
+resource "aws_iam_role_policy" "streamalert_rule_processor_cloudwatch" {
+  name = "CloudwatchWriteLogs"
+  role = "${aws_iam_role.streamalert_rule_processor_role.id}"
+
+  policy = "${data.aws_iam_policy_document.alert_processor_cloudwatch.json}"
+}
+
 // IAM Role Policy: Allow the Rule Processor to invoke the Alert Processor
 resource "aws_iam_role_policy" "streamalert_rule_processor_lambda" {
-  name = "${var.prefix}_${var.cluster}_streamalert_rule_processor_invoke_alert_proc"
+  name = "LambdaInvokeAlertProcessor"
```

Comment:
The `name` of role policy should be unique. If it is not unique, this policy will be shared with multiple IAM roles when there are multiple clusters in the environment. When the user destroy one cluster for some reason, this policy might be deleted, although other IAM roles from other clusters are still referring to this policy.
I tested it by creating two simple `aws_iam_role_policy` with same `name`. Policies can be created successfully, however, `terraform destroy` throws error says `NoSuchEntity` when it tries to destroy second iam role policy.

Reply:
Same comment is applied to following `aws_iam_role_policy` changes in this tf file.

Reply:
@chunyong-lin can you explain your testing a little more? Did you create two role policies in the same role with the same name? or different roles? they are namespaced per role, so it shouldn't conflict

Reply:
this did just lead me to finding another bug though in the naming of the Firehose role

Reply:
I created two role policies in the same role with the same name. 
Example:
```
resource "aws_iam_role_policy" "foo" {
  name   = "role_policy_name_baz"
  role   = "${aws_iam_role.pan.id}"
  policy = "${data.aws_iam_policy_document.foo.json}"
}

resource "aws_iam_role_policy" "bar" {
  name   = "role_policy_name_baz"
  role   = "${aws_iam_role.pan.id}"
  policy = "${data.aws_iam_policy_document.bar.json}"
}
```
Creation was successful, but deletion complaint `NoSuchEntity` when it tries to destroy second policy.